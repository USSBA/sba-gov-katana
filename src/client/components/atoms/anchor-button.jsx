import React from 'react';
import { Button } from 'react-bootstrap'

class AnchorButton extends React.Component {

  scroll() {
    let element = document.getElementById(this.props.anchor);
    element.scrollIntoView({
      block: "start",
      behavior: "smooth"
    });
  }

  render() {
    return (
      <Button block className={ this.props.className } onClick={ this.scroll.bind(this) }>
        {this.props.text}
      </Button>
    )
  }
}

export default AnchorButton;
