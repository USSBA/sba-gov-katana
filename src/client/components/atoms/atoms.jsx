import CaptionText from "./caption-text"
