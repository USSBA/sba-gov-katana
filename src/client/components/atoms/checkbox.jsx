import React from 'react';
import Checkbox from 'rc-checkbox';
import './checkbox.scss';

class CheckBox extends React.Component {

  render() {
    return (
      <div  className="rc-checkbox-container">
        <label className="rc-checkbox-label">
        <Checkbox/>
          here is the label
        </label>

      </div>


      );
  }
}

export default CheckBox;