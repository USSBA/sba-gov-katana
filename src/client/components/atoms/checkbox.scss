$money-green: #609F00;

.rc-checkbox {
    white-space: nowrap;
    cursor: pointer;
    outline: none;
    display: inline-block;
    position: relative;
    width: 20px;
    height: 18px;
    left: -35px;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
}
//.rc-checkbox:hover .rc-checkbox-inner,
//.rc-checkbox-input:focus + .rc-checkbox-inner {
//    border-color: $money-green;
//}



.rc-checkbox-inner {
    position: relative;
    top: 0;
    left: 0;
    display: inline-block;
    width: 19px;
    height: 18px;
    border-width: 1px;
    border-style: solid;
    border-radius: 0px;
    border-color: #d9d9d9;
    background-color: #ffffff;
    transition: border-color 0.3s cubic-bezier(0.68, -0.55, 0.27, 1.55), background-color 0.3s cubic-bezier(0.68, -0.55, 0.27, 1.55);
}
.rc-checkbox-inner:after {
    transform: rotate(45deg);
    position: absolute;
    left: 5px;
    top: 2px;
    margin: auto;
    width: 7px;
    height: 10px;
    border: 3px solid #ffffff;
    border-top: 0;
    border-left: 0;
    content: ' ';
    animation-timing-function: cubic-bezier(0.68, -0.55, 0.27, 1.55);
    animation-duration: 0.3s;
    animation-name: amCheckboxOut;
}
.rc-checkbox-input {
    position: absolute;
    left: -15px;
    z-index: 9999;
    cursor: pointer;
    opacity: 0;
    top: -17px;
    bottom: 0;
    right: 0;
    width: 350px;
    height: 47px;
}
/* 选中状态 */
.rc-checkbox-checked:hover .rc-checkbox-inner {
    border-color: $money-green;
}
.rc-checkbox-checked .rc-checkbox-inner {
    border-color: $money-green;
    background-color: $money-green;
}
.rc-checkbox-checked .rc-checkbox-inner:after {
    transform: rotate(45deg);
    position: absolute;
    left: 5px;
    top: 2px;
    margin: auto;
    width: 7px;
    height: 10px;
    border: 3px solid #ffffff;
    border-top: 0;
    border-left: 0;
    content: ' ';
    animation-timing-function: cubic-bezier(0.68, -0.55, 0.27, 1.55);
    animation-duration: 0.3s;
    animation-name: amCheckboxOut;
}
.rc-checkbox-disabled.rc-checkbox-checked:hover .rc-checkbox-inner {
    border-color: #d9d9d9;
}
.rc-checkbox-disabled.rc-checkbox-checked .rc-checkbox-inner {
    background-color: #f3f3f3;
    border-color: #d9d9d9;
}
.rc-checkbox-disabled.rc-checkbox-checked .rc-checkbox-inner:after {
    animation-name: none;
    border-color: #cccccc;
}
.rc-checkbox-disabled:hover .rc-checkbox-inner {
    border-color: #d9d9d9;
}
.rc-checkbox-disabled .rc-checkbox-inner {
    border-color: #d9d9d9;
    background-color: #f3f3f3;
}
.rc-checkbox-disabled .rc-checkbox-inner:after {
    animation-name: none;
    border-color: #f3f3f3;
}
.rc-checkbox-disabled .rc-checkbox-inner-input {
    cursor: default;
}
@keyframes amCheckboxIn {
    0% {
        opacity: 0;
        transform-origin: 50% 50%;
        transform: scale(0, 0) rotate(45deg);
    }
    100% {
        opacity: 1;
        transform-origin: 50% 50%;
        transform: scale(1, 1) rotate(45deg);
    }
}
@keyframes amCheckboxOut {
    0% {
        opacity: 1;
    }
    100% {
        opacity: 0;
    }
}

//CUSTOM STYLES FOR CHECKBOX

.rc-checkbox-container {
    position: relative;
    width: 360px;
    height: 48px;
    border: 1px solid #C4C4C4;
    margin-bottom: 10px;
}

.rc-checkbox-container-checked {
    position: relative;
    width: 360px;
    height: 48px;
    border: 1px solid #666666;
    margin-bottom: 10px;
}

.rc-checkbox-container-focused {
    position: relative;
    width: 360px;
    height: 48px;
    border: 1px solid $money-green;
    margin-bottom: 10px;
}



.rc-checkbox-label {
    left: 53px;
    display: inline-block;
    font-size: 18px;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
}
