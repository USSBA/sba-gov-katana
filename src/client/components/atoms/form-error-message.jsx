import React from 'react';
import formHelperStyles from './form-helpers.scss';

class FormErrorMessage extends React.Component {
  render() {
    return (
      <p className={formHelperStyles.errorText}>
          {this.props.errorText}
      </p>
    );
  }
}
export default FormErrorMessage;
