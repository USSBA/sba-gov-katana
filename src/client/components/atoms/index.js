import ArrowButton from "./arrow-button/arrow-button.jsx";
import ButtonBase from "./button-base/button-base.jsx";
import CaptionText from "./caption-text/caption-text.jsx";
import Carousel from "./carousel/carousel.jsx";
import Checkbox from "./checkbox/checkbox.jsx";
import DecorativeDash from "./decorative-dash/decorative-dash.jsx";
import ExtraLargeTitleText from "./extra-large-title-text/extra-large-title-text.jsx";
import FormErrorMessage from "./form-error-message/form-error-message.jsx";
import HamburgerIcon from "./hamburger-icon/hamburger-icon.jsx";
import LargeGreySecondaryButton from "./large-grey-secondary-button/large-grey-secondary-button.jsx";
import LargeInversePrimaryButton from "./large-inverse-primary-button/large-inverse-primary-button.jsx";
import LargeInverseSecondaryButton from "./large-inverse-secondary-button/large-inverse-secondary-button.jsx";
import LargePrimaryButton from "./large-primary-button/large-primary-button.jsx";
import LargeSecondaryButton from "./large-secondary-button/large-secondary-button.jsx";
import MainLogo from "./main-logo/main-logo.jsx";
import Multiselect from "./multiselect/multiselect.jsx";
import NewsletterSmallPrimaryButton from "./newsletter-small-primary-button/newsletter-small-primary-button.jsx";
import PageLink from "./page-link/page-link.jsx";
import PdfIcon from "./icons/pdf.jsx";
import ProgressBar from "./progress-bar/progress-bar.jsx";
import Radio from "./radio/radio.jsx";
import SectionLink from "./section-link/section-link.jsx";
import SimpleSelect from "./simple-select/simple-select.jsx";
import SmallGreySecondaryButton from "./small-grey-secondary-button/small-grey-secondary-button.jsx";
import SmallGreySecondaryFormButton from "./small-grey-secondary-form-button/small-grey-secondary-form-button.jsx";
import SmallIcon from "./small-icon/small-icon.jsx";
import SmallInversePrimaryButton from "./small-inverse-primary-button/small-inverse-primary-button.jsx";
import SmallInverseSecondaryButton from "./small-inverse-secondary-button/small-inverse-secondary-button.jsx";
import SmallPrimaryButton from "./small-primary-button/small-primary-button.jsx";
import SmallPrimaryFormButton from "./small-primary-form-button/small-primary-form-button.jsx";
import SmallSecondaryButton from "./small-secondary-button/small-secondary-button.jsx";
import SocialMediaLink from "./social-media-link/social-media-link.jsx";
import SubtitleText from "./subtitle-text/subtitle-text.jsx";
import TextInput from "./text-input/text-input.jsx";
import TextArea from "./textarea/textarea.jsx";
import UtilityLink from "./utility-link/utility-link.jsx";
import DocumentType from "./document-type/document-type.jsx";
import ValidationIcon from "./validation-icon/validation-icon.jsx";
import SearchIcon from "./icons/search.jsx";
import ApplyButton from "./apply-button/apply-button.jsx";
import CaretIcon from "./icons/caret-down.jsx";
import CloseIcon from "./icons/close.jsx";
import HomeIcon from "./icons/home.jsx";
import FailureIcon from "./icons/failure.jsx";
import SuccessIcon from "./icons/success.jsx";

export { ArrowButton, ButtonBase, CaptionText, Carousel, Checkbox, DecorativeDash, ExtraLargeTitleText, FormErrorMessage, HamburgerIcon, LargeGreySecondaryButton, LargeInversePrimaryButton, LargeInverseSecondaryButton, LargePrimaryButton, LargeSecondaryButton, MainLogo, Multiselect, NewsletterSmallPrimaryButton, PageLink, PdfIcon, ProgressBar, Radio, SectionLink, SimpleSelect, SmallGreySecondaryButton, SmallIcon, SmallInversePrimaryButton, SmallInverseSecondaryButton, SmallPrimaryButton, SmallSecondaryButton, SocialMediaLink, SubtitleText, TextInput, TextArea, UtilityLink, ValidationIcon, SearchIcon, ApplyButton, DocumentType, CaretIcon, CloseIcon, HomeIcon, FailureIcon, SuccessIcon };
