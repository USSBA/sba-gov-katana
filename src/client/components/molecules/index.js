import Breadcrumb from "./breadcrumb/breadcrumb.jsx";
import ButtonCta from "./button-cta/button-cta.jsx";
import CallToAction from "./call-to-action/call-to-action.jsx";
import Callout from "./callout/callout.jsx";
import Card from "./card/card.jsx";
import CardGrid from "./card-grid/card-grid.jsx";
import ContactCardLookup from "./contact-card-lookup/contact-card-lookup.jsx";
import DocumentArticle from "./document-article/document-article.jsx";
import DocumentCard from "./document-card/document-card.jsx";
import DocumentLookup from "./document-lookup/document-lookup.jsx";
import DropdownMenu from "./dropdown-menu/dropdown-menu.jsx";
import FeaturedCallout from "./featured-callout/featured-callout.jsx";
import FeedbackForm from "./feedback-form/feedback-form.jsx";
import FormPageButtons from "./form-page-buttons/form-page-buttons.jsx";
import GoogleTranslate from "./google-translate/google-translate.jsx";
import ImageSection from "./image-section/image-section.jsx";
import Lookup from "./lookup/lookup.jsx";
import PageLinkGroup from "./page-link-group/page-link-group.jsx";
import Paginator from "./paginator/paginator.jsx";
import ParagraphPlaceholder from "./paragraph-placeholder/paragraph-placeholder.jsx";
import PreviousNextSection from "./previous-next/previous-next.jsx";
import QuickLinks from "./quick-links/quick-links.jsx";
import ReadMore from "./readmore-section/readmore-section.jsx";
import ResourceCard from "./resource-card/resource-card.jsx";
import SbicLookup from "./sbic-lookup/sbic-lookup.jsx";
import SearchBar from "./search-bar/search-bar.jsx";
import SectionHeader from "./section-header/section-header.jsx";
import SbaModal from "./sba-modal/sba-modal.jsx"
import SimpleCta from "./simple-cta/simple-cta.jsx";
import SmallInverseCta from "./small-inverse-cta/small-inverse-cta.jsx";
import StyleGrayBackground from "./style-gray-background/style-gray-background.jsx";
import SubMenu from "./sub-menu/sub-menu.jsx";
import SubsectionHeader from "./subsection-header/subsection-header.jsx";
import SuretyLookup from "./surety-lookup/surety-lookup.jsx";
import TenStepsNav from "./ten-steps-nav/ten-steps-nav.jsx";
import TenStepsSection from "./ten-steps-section/ten-steps-section.jsx";
import TextReadMoreSection from "./text-readmore-section/text-readmore-section.jsx";
import TileCollection from "./tile-collection/tile-collection.jsx";
import TitleBox from "./title-box/title-box.jsx";
import TitleSection from "./title-section/title-section.jsx";
import TextSection from "./text-section/text-section.jsx";

export { Breadcrumb, ButtonCta, CallToAction, Callout, Card, CardGrid, ContactCardLookup, DocumentArticle, DocumentCard, DocumentLookup, DropdownMenu, FeaturedCallout, FeedbackForm, FormPageButtons, GoogleTranslate, ImageSection, Lookup, PageLinkGroup, Paginator, ParagraphPlaceholder, PreviousNextSection, QuickLinks, ReadMore, ResourceCard, SbicLookup, SearchBar, SectionHeader, SimpleCta, SmallInverseCta, StyleGrayBackground, SubMenu, SubsectionHeader, SuretyLookup, TenStepsNav, TenStepsSection, TextReadMoreSection, TextSection, TileCollection, TitleBox, TitleSection, SbaModal };
