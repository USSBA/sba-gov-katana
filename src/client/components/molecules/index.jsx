import Address from './address/address'
import AuthorCard from './author-card/author-card'
import BizzieChatbotWidget from './bizzie-chatbot-widget/bizzie-chatbot-widget'
import Breadcrumb from './breadcrumb/breadcrumb'
import ButtonCta from './button-cta/button-cta'
import CallToAction from './call-to-action/call-to-action'
import Card from './card/card'
import CardGrid from './card-grid/card-grid'
import ContactCard from './contact-card/contact-card'
import ContactCardLookup from './contact-card-lookup/contact-card-lookup'
import DetailCard from './detail-card/detail-card'
import DistrictOfficePreviousNextSection from './district-office-previous-next/district-office-previous-next'
import FeedbackForm from './feedback-form/feedback-form'
import FormPageButtons from './form-page-buttons/form-page-buttons'
import GoogleTranslate from './google-translate/google-translate'
import LenderResultCard from './lender-result-card/lender-result-card'
import MainLoader from './main-loader/main-loader'
import MenuTile from './menu-tile/menu-tile'
import MenuTileWithLinks from './menu-tile-with-links/menu-tile-with-links'
import NewsletterForm from './newsletter-form/newsletter-form'
import NoResultsSection from './no-results-section/no-results-section'
import NumberedCalloutBlock from './numbered-callout-block/numbered-callout-block'
import PageLinkGroup from './page-link-group/page-link-group'
import Paginator from './paginator/paginator'
import PhoneNumber from './phone-number/phone-number'
import PreviousNextSection from './previous-next/previous-next'
import QuickLinks from './quick-links/quick-links'
import ReadMore from './readmore/readmore'
import ReadMoreSection from './readmore-section/readmore-section'
import RemoveMainLoader from './main-loader/remove-main-loader'
import SbaModal from './sba-modal/sba-modal'
import SbaNewsModal from './news-modal/news-modal'
import SearchBar from './search-bar/search-bar'
import SuggestedRouteCard from './suggested-route-card/suggested-route-card'
import LongScrollNav from './long-scroll-nav/long-scroll-nav'
import TitleSection from './title-section/title-section'
import Typeahead from './typeahead/typeahead'

export {
  Address,
  AuthorCard,
  BizzieChatbotWidget,
  Breadcrumb,
  ButtonCta,
  CallToAction,
  Card,
  CardGrid,
  ContactCard,
  ContactCardLookup,
  DetailCard,
  DistrictOfficePreviousNextSection,
  FeedbackForm,
  FormPageButtons,
  GoogleTranslate,
  LenderResultCard,
  MainLoader,
  MenuTile,
  MenuTileWithLinks,
  NewsletterForm,
  NoResultsSection,
  NumberedCalloutBlock,
  PageLinkGroup,
  Paginator,
  PhoneNumber,
  PreviousNextSection,
  QuickLinks,
  ReadMore,
  ReadMoreSection,
  RemoveMainLoader,
  SbaModal,
  SbaNewsModal,
  SearchBar,
  SuggestedRouteCard,
  LongScrollNav,
  TitleSection,
  Typeahead
}
