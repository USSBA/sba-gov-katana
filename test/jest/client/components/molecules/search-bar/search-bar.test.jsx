/* eslint-env jest */
import React from 'react' // eslint-disable-line no-unused-vars
import { shallow } from 'enzyme'

import { SearchBar } from 'molecules'
import { Link, TextInput } from 'atoms'

describe('SearchBar', () => {
  test('should render the happy path', () => {
    const component = shallow(<SearchBar />)
    expect(component).toMatchSnapshot()
  })

  test('should expand and then submit search', () => {
    const component = shallow(<SearchBar />)
    let mockSearcher = jest.fn()
    SearchBar.prototype.executeSearch = mockSearcher
    component.find(Link).simulate('click', { preventDefault: _ => true })
    expect(component.state('expanded')).toEqual(true)
    component
      .find(TextInput)
      .simulate('change', { target: { value: 'business' }, preventDefault: _ => true })
    expect(component.state('searchValue')).toEqual('business')
    component.find('i#search-button').simulate('click', { preventDefault: _ => true })
    expect(mockSearcher.mock.calls.length).toBe(1)
    expect(mockSearcher.mock.calls[0][0]).toBe('/mysearch?q=business')
  })
})
