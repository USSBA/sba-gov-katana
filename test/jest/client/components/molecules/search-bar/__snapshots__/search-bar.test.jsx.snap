// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render the happy path 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Link
    aria-label="toggle search bar"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <i
      alt="toggle search bar"
      aria-hidden="true"
      className="fa fa-search"
      id="search-toggle-link"
    />
  </Link>,
  "nodes": Array [
    <Link
      aria-label="toggle search bar"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <i
        alt="toggle search bar"
        aria-hidden="true"
        className="fa fa-search"
        id="search-toggle-link"
      />
    </Link>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchBar
        onExpand={[Function]}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchBar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onExpand": [Function],
        },
        "refs": Object {},
        "state": Object {
          "expanded": false,
          "searchValue": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Link
          aria-label="toggle search bar"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <i
            alt="toggle search bar"
            aria-hidden="true"
            className="fa fa-search"
            id="search-toggle-link"
          />
        </Link>,
        "_debugID": 2,
        "_renderedOutput": <Link
          aria-label="toggle search bar"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <i
            alt="toggle search bar"
            aria-hidden="true"
            className="fa fa-search"
            id="search-toggle-link"
          />
        </Link>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBar
    onExpand={[Function]}
  />,
}
`;
